src/node/desktop/src/main/preferences/preferences.ts

```diff
@@ -0,0 +1,37 @@
+import { UserDefaultTypes } from 'electron';
+import { homedir } from 'os';
+import MacPreferences from './mac-preferences';
+import FilePreferences from './file-preferences';
+
+export interface PlatformPreferences {
+  getValue(
+    key: string,
+    type?: keyof UserDefaultTypes,
+  ): string | number | boolean | unknown[] | Record<string, unknown> | undefined;
+  setValue(key: string, type: keyof UserDefaultTypes, value: string): void;
+}
+
+const isMacOS = process.platform === 'darwin';
+
+/**
+ * MacOS stores periods as · in defaults. When adding a new key, ensure that the key is valid
+ * on all platforms.
+ */
+export const preferenceKeys = {
+  fontFixedWidth: isMacOS ? 'font·fixedWidth' : 'General.font.fixedWidth',
+};
+
+const getPreferenceManager = () => {
+  switch (process.platform) {
+    case 'darwin':
+      return new MacPreferences();
+    case 'win32':
+      return new FilePreferences(`${homedir()}\\AppData\\Roaming\\RStudio\\desktop.ini`);
+    case 'linux':
+      return new FilePreferences(`${homedir()}/.config/RStudio/desktop.ini`);
+    default:
+      throw Error('unsupported platform');
+  }
+};
+
+export default getPreferenceManager;
```

Comment:
I think you could also just export an object called `preferenceManager` and initialize it when this module is loaded. IMHO that's marginally cleaner; do you have a strong opinion?

Reply:
Yes, I think that would be a cleaner option! 👍